Transaction statements not visible in pg_stat_statements view

kritika-0601 opened the pull request:
Currently Babelfish does not show transaction queries inside the pg_stat_statements table. After my change now it will be visible in the view.
pg_stat_statements displays planning and execution statistics for all SQL statements executed by a server. Some of the statements are not being tracked by it. Mainly because subsequent processutility hooks after bbf_processutilty are not being called, what happens is inside bbf_processUtility, babelfish specific function is called and it does not trigger the subsequent hooks due to which query statistics are not being stored in the table.
Created a hook for that and called it inside relevant ProcessUtility function.

PR link for engine code: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/118

### Test Scenarios Covered ###
* **Use case based -**
N/A

* **Boundary conditions -**
N/A

* **Arbitrary inputs -**
N/A

* **Negative test cases -**
N/A

* **Minor version upgrade tests -**
N/A

* **Major version upgrade tests -**
N/A

* **Performance tests -**
N/A

* **Tooling impact -**
N/A

* **Client tests -**
N/A


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
retrigger tests

Commit:
retrigger tests

surendravishnoi commented on contrib/babelfishpg_tsql/src/hooks.c:
```diff
@@ -342,11 +347,22 @@ UninstallExtendedHooks(void)
+	if (NestedTranCount > 0 || (sql_dialect == SQL_DIALECT_TSQL && !IsTransactionBlockActive()))
+	{
+		PLTsqlProcessTransaction(parsetree, params, qc);
+		return;
```
return is not required

kritika-0601 committed:
Update major-version-upgrade.yml

kritika-0601 committed:
Merge branch 'babelfish-for-postgresql:BABEL_3_X_DEV' into pg_stat_statements-transactions

surendravishnoi commented on contrib/babelfishpg_tsql/src/pl_handler.c:
```diff
@@ -2358,15 +2358,15 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
-				}
-				break;
-			}
+		// case T_TransactionStmt:
```
please remove the code no longer required

surendravishnoi commented on contrib/babelfishpg_tsql/src/hooks.h:
```diff
@@ -3,6 +3,7 @@
 #include "postgres.h"
 #include "catalog/catalog.h"
 #include "parser/analyze.h"
+#include "tcop/cmdtag.h"
```
why do we need this header for? Ideally required headers should be added to .c files only

kritika-0601 replied:
I've added this because in this function PLTsqlProcessTransaction's parameters we have this QueryCompletion, for that we need this header.

surendravishnoi commented on contrib/babelfishpg_tsql/src/hooks.c:
```diff
@@ -342,11 +347,22 @@ UninstallExtendedHooks(void)
 /*****************************************
  * 			Hook Functions
  *****************************************/
+static void
```
we should have boolean return type to control engine side execution based on whether we call PLTsqlProcessTransaction or not

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Update major-version-upgrade.yml

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
retrigger tests

Commit:
Transaction statements not visible in pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty are not being called.
Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Merge branch 'BABEL_3_X_DEV' into pg_stat_statements-transactions

Commit:
retrigger tests
